wasmbinding/message_plugin.go

```diff
@@ -334,6 +361,170 @@ func (m *CustomMessenger) performSubmitAdminProposal(ctx sdk.Context, contractAd
 	return response, nil
 }
 
+// createDenom creates a new token denom
+func (m *CustomMessenger) createDenom(ctx sdk.Context, contractAddr sdk.AccAddress, createDenom *bindings.CreateDenom) ([]sdk.Event, [][]byte, error) {
+	err := PerformCreateDenom(m.TokenFactory, m.Bank, ctx, contractAddr, createDenom)
+	if err != nil {
+		return nil, nil, sdkerrors.Wrap(err, "perform create denom")
+	}
+	return nil, nil, nil
+}
+
+// PerformCreateDenom is used with createDenom to create a token denom; validates the msgCreateDenom.
+func PerformCreateDenom(f *tokenfactorykeeper.Keeper, b *bankkeeper.BaseKeeper, ctx sdk.Context, contractAddr sdk.AccAddress, createDenom *bindings.CreateDenom) error {
+	if createDenom == nil {
+		return wasmvmtypes.InvalidRequest{Err: "create denom null create denom"}
+	}
```

Comment:
I suggest removing this check since this function is only called when CreateDenom is not nil anyway.